### ISSUE TITLE
Add Block Diffusion Attention Mask to examples

### ISSUE THREAD

**Issue by Skylion007** [pytorch-labs/attention-gym#126]
![Image](https://github.com/user-attachments/assets/a8f96902-80dc-4d7c-a6e8-a41c6df725cf)
Implement the Block Diffusion Hybrid Attention Kernel from https://arxiv.org/abs/2503.09573 . Disclaimer I am an author. It's a simple, illustrative, and efficient kernel that can achieve >5X speedup over SDPA due to the sparsity of the mask. Therefore, probably a good example to highlight flex attention for performance reasons.

**Comment by drisspg** [comment]
@Skylion007 you want to contribute it? 😊

**Comment by drisspg** [comment]
landed: https://github.com/meta-pytorch/attention-gym/pull/217
